src/terrain/terrain.cpp

```diff
@@ -452,6 +452,21 @@ Terrain::init_grass() {
 }
 
 
+void
+Terrain::grow_grass_high(std::set<Tile*> all_grass) {
+    helper::grow_grass_recursive<
+        helper::edge_detector_high, helper::getter_high, helper::setter_high>(
+        *this, all_grass);
+}
+
+void
+Terrain::grow_grass_low(std::set<Tile*> all_grass) {
+    helper::grow_grass_recursive<
+        helper::edge_detector_low, helper::getter_low, helper::setter_low>(
+        *this, all_grass
+    );
+}
+
```

Comment:
Why did you move these back and un-inline them?

Reply:
i, um uh, couldn't um, get it compile...

Reply:
I can't import helper in terrain and import terrain in helper

Reply:
- Move to the top of the file
- Make it just a plain static function
- Mark as inline
- PROFIT

Reply:
If you can figure out how to do this with out eight forward declarations be my guest.

Reply:
> terrain.hpp? then I would need to import terrain_helper or make a forward definition of helper::edge_detector

No, `terrain.cpp`. Its only called from within that file.

> ? static? This is a stand alone function. I think this is the same as inline.

No you need both they don't imply each other.